src/Task/MapRoute.pm

```diff
@@ -271,8 +271,25 @@ sub iterate {
 
 	} elsif ( $portals_lut{"$self->{mapSolution}[0]{map} $self->{mapSolution}[0]{pos}{x} $self->{mapSolution}[0]{pos}{y}"}{source} ) {
 		# This is a portal solution
+		
+		
+		if ( distance($self->{actor}{pos_to}, $self->{mapSolution}[0]{pos}) == 0 ) {
+				return unless (timeOut($timeout{ai_portal_wait}));
```

Comment:
`ai_portal_wait` defaults to 0.5 seconds. This could cause deleted portals and re-routes on slow servers, until eventually the bot can't go anywhere because there aren't any portals left which the bot can use.